### ISSUE TITLE
Offer an official plexdrive docker image

### ISSUE THREAD

**Issue by Rojikku** [dweidenfeld/plexdrive#414]
Plexdrive is a nice piece of software, but my environment is containerized, so I can't use it directly..
Previously, wiserain had maintained an unofficial [docker image](https://hub.docker.com/r/wiserain/plexdrive) which I had used for my use case, but I've noted this is no longer maintained.
This image has over a million downloads. I think this is sufficient to speak of the popularity of a docker image version of plexdrive.

With the current functionality of github CI pipelines and direct support for containerization, I don't think it would be very difficult to maintain once it was setup. There's plenty of examples, including the above which has a linked github repository, which could be used to assist in the initial setup.

**Comment by felixbuenemann** [comment]
How useful is a docker image? You also need to run something like plexmediaserver to stream the media from the container.

**Comment by Skyfay** [comment]
The advantage of docker is that you have a docker compose file where you have all services such as plexdrive, plex (other media servers) inside. The backup is easy and one run and one upgrade makes Docker much easier and more efficient.

The player for plexdrive does not necessarily have to be used as a Docker image, it is also possible without it because the files are placed on the host.
Everything has been in Docker for me for a while now, which makes the administrator and a restore to other servers much easier. It would be a pity if I always had to install and configure the whole thing directly on the host with Plexdrive.

I would also find a Docker image very cool!

**Comment by Rojikku** [comment]
I've had no problems passing fuse to the container and then mounting my plexdrive to my host. I then have another container, https://hub.docker.com/r/jellyfin/jellyfin/ that also connects to the house to read those files. There's other, more secure ways you can do this as well, but that's all on the user end and outside of the docker image itself.
In my use case I use kubernetes, and I can use some other services to securely pass fuse to the plexdrive. I could theoretically also run plexdrive as a sidecar to avoid mounting to the host, but I have not.

The primary uses for a docker container are to be able to easily spin up services in these setups. I have additional services setup that make pull requests to my repository when containers are updated, so this also means plexdrive can autoupdate relatively easily.
Take for example my running [rclone setup](https://github.com/Rojikku/k3s/blob/main/cluster/apps/sonarr/rclone/helm-release.yaml).
For my use case I do require a HelmChart, but, I can easily do that [myself](https://github.com/Rojikku/charts/tree/main/charts/rclone).
I can, with relatively few commands, spin up my entire home cluster with all the services I run, restore from backups, and deal with hardware failures and such. I get discord notifications on automatic pull requests to update my containers, and the underlying systems (flux) handle everything else once I hit merge.

Advantages to a developer being that, generally speaking, as long as your container works and is self-contained (I.E. not requiring a postgresql database or something), you can generally expect any bug reports relating to the docker image to be in a replicable environment, since you would know the application's environment was setup with correct dependencies.

TL;DR: Lots of cool automation setups, environment is setup correctly since devs write the build of the image, and can run in docker, kubernetes, and other cloud environments.
Why users setup cloud environments, no idea. I just do it because I enjoy it.

**Comment by Skyfay** [comment]
The Docker image got an update!

https://hub.docker.com/r/wiserain/plexdrive
